src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Generated/Models/IndexAction.cs

```diff
@@ -0,0 +1,56 @@
+// <auto-generated>
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+// </auto-generated>
+
+namespace Microsoft.Azure.Search.Models
+{
+    using Newtonsoft.Json;
+    using System.Linq;
+
+    public partial class IndexAction
+    {
+        /// <summary>
+        /// Initializes a new instance of the IndexAction class.
+        /// </summary>
+        public IndexAction()
+        {
+            CustomInit();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the IndexAction class.
+        /// </summary>
+        /// <param name="actionType">Possible values include: 'upload',
+        /// 'merge', 'mergeOrUpload', 'delete'</param>
+        public IndexAction(IndexActionType? actionType = default(IndexActionType?), object document = default(object))
+        {
+            ActionType = actionType;
+            Document = document;
+            CustomInit();
+        }
+
+        /// <summary>
+        /// An initialization method that performs custom operations like setting defaults
+        /// </summary>
+        partial void CustomInit();
+
+        /// <summary>
+        /// Gets or sets possible values include: 'upload', 'merge',
+        /// 'mergeOrUpload', 'delete'
+        /// </summary>
+        [JsonProperty(PropertyName = "actionType")]
+        public IndexActionType? ActionType { get; set; }
+
+        /// <summary>
+        /// </summary>
+        [JsonProperty(PropertyName = "document")]
```

Comment:
Below are the serializer changes. The 'object' in *Document are de/serialized using the existing DocumentConverter (if TDoc == Document ) or the type specific one otherwise. 

            settings.Converters.Add(new SearchDocumentConverter<TDoc>());
            settings.Converters.Add(new SuggestDocumentConverter<TDoc>());
            settings.Converters.Add(new LookupDocumentConverter<TDoc>());
            settings.Converters.Add(new DocumentConverter());

Reply:
We already have the serializer for Document type and yes it still works.